src/dependency-manager/src/spec/common/scaling-v1.ts

```diff
@@ -0,0 +1,27 @@
+import { Type } from 'class-transformer';
+import { IsInstance, IsOptional } from 'class-validator';
+import { ValidatableConfig } from '../base-spec';
+
+export class ScalingMetricsSpecV1 extends ValidatableConfig {
+  @IsOptional({ always: true })
+  @Type(() => String)
+  cpu?: string;
+
+  @IsOptional({ always: true })
+  @Type(() => String)
+  memory?: string;
+}
+
+export class ScalingSpecV1 extends ValidatableConfig {
+  @IsOptional({ always: true })
+  @Type(() => String)
+  min_replicas?: string;
+
+  @IsOptional({ always: true })
+  @Type(() => String)
+  max_replicas?: string;
+
+  @IsOptional({ always: true })
```

Comment:
Are any of these optional assuming they added a scaling block? I think we should require. I could see cpu or memory being optional, but at least one has to be set. Unsure if we should have defaults for those.